close_test.go

```diff
@@ -192,22 +211,58 @@ func TestRaceExchangesWithClose(t *testing.T) {
 				if closed.Load() {
 					return
 				}
+
+				ctx, cancel := NewContext(testutils.Timeout(time.Second))
+				defer cancel()
+
 				if err := c.Ping(ctx, ts.HostPort()); err != nil {
 					return
 				}
+
 				if closed.Load() {
 					return
 				}
+
 				raw.Call(ctx, c, ts.HostPort(), server.ServiceName(), "dummy", nil, nil)
 			}()
 		}
 
 		// Now try to close the channel, it should block since there's active exchanges.
 		server.Close()
 		closed.Store(true)
-		assert.Equal(t, ChannelStartClose, ts.Server().State(), "Server should be in StartClose")
-		closed.Store(true)
 
+		// n.b. As it's shutting down, server state can be in any of the
+		//      outlined states below. It doesn't matter which specific state
+		//      it's in, as long as we're verifying that it's at least in the
+		//      process of shutting down.
```

Comment:
I don't think this is true -- we explicitly expect the channel state to be in `ChannelStartClose` because there's an inbound call that's not complete.

See:
The channel stays in ChannelStartClose till all connections are past connectionInboundClosed:
https://github.com/aiven/tchannel-go/blob/v1.22.0/channel.go#L794

Connections don't transition to `connectionInboundClosed` till the inbound exchange count is 0:
https://github.com/aiven/tchannel-go/blob/v1.22.0/connection.go#L818

The inbound exchange count is the number of pending calls, and until the below `close(completeCall)`, the server side echo handler will block.